slmigrate/arghandler.py

```diff
@@ -3,50 +3,43 @@
-    parser.add_argument("--" + constants.source_db_arg, "--sourcedb", help="The name of the source directory when performing intra-databse migration", action="store", default=constants.source_db)
-    return parser
-
+def parse_arguments():
+    parser = argparse.ArgumentParser(prog='slmigrate')
+
+    parent_parser = argparse.ArgumentParser(add_help=False)
+    parent_parser.add_argument("--" + constants.tag.arg, "--tags", "--tagingestion", "--taghistory", help="Migrate tags and tag histories", action="store_true", )
+    parent_parser.add_argument("--" + constants.opc.arg, "--opcua", "--opcuaclient", help="Migrate OPCUA sessions and certificates", action="store_true")
+    parent_parser.add_argument("--" + constants.fis.arg, "--file", "--files", help="Migrate ingested files", action="store_true")
+    parent_parser.add_argument("--" + constants.testmonitor.arg, "--test", "--tests", "--testmonitor", help="Migrate Test Monitor data", action="store_true")
+    parent_parser.add_argument("--" + constants.asset.arg, "--assets", help="Migrate asset utilitization and calibration data", action="store_true")
+    parent_parser.add_argument("--" + constants.repository.arg, "--repo", help="Migrate packages and feeds", action="store_true")
+    parent_parser.add_argument("--" + constants.alarmrule.arg, "--alarms", "--alarm", help="Migrate Tag alarm rules", action="store_true")
+    parent_parser.add_argument("--" + constants.userdata.arg, "--ud", help="Migrate user data", action="store_true")
+    parent_parser.add_argument("--" + constants.notification.arg, "--notifications", help="Migrate notifications strategies, templates, and groups", action="store_true")
+    parent_parser.add_argument("--" + constants.states.arg, "--state", help="Migrate system states", action="store_true")
+    parent_parser.add_argument("--" + constants.migration_arg, "--directory", "--folder", help="Specify the directory used for migrated data", action="store", default=constants.migration_dir)
+    parent_parser.add_argument("--" + constants.source_db_arg, "--sourcedb", help="The name of the source directory when performing intra-databse migration", action="store", default=constants.source_db)
+
+    commands = parser.add_subparsers(dest='action')
+    commands.add_parser(constants.capture_arg, help="capture is used to pull data and settings off SystemLink server", parents=[parent_parser])
+    commands.add_parser(constants.restore_arg, help="restore is used to push data and settings to a clean SystemLink server. ", parents=[parent_parser])
+  
+    commands.add_parser(constants.thdbbug.arg, help="Migrate tag history data to the correct MongoDB to resolve an issue introduced in SystemLink 2020R2 when using a remote Mongo instance. Use --sourcedb to specify a source database. admin is used if none is specfied",)
 
-def handle_unallowed_args(arguments):
```

Comment:
I'm assuming this function was removed because sub parsers should do this for "free"? That's ok except for its still being called from systemlinkmigrate.py, which leads to an exception.

```
PS C:\Users\markblack\Documents\GitHub\systemlink-migration-sandbox> python .\systemlinkmigrate.py capture --test
Traceback (most recent call last):
  File ".\systemlinkmigrate.py", line 10, in <module>
    arghandler.handle_unallowed_args(arguments)
AttributeError: module 'slmigrate.arghandler' has no attribute 'handle_unallowed_args'

Reply:
Sorry about that oversight.  I was so focused on getting the tests to pass that I forgot to update the slmigrate.py script.  I've done that now.